scipy/interpolate/_pade.py

```diff
@@ -52,15 +54,17 @@ def pade(an, m, n=None):
     if N > len(an)-1:
         raise ValueError("Order of q+p <m+n> must be smaller than len(an).")
     an = an[:N+1]
-    Akj = eye(N+1, n+1, dtype=an.dtype)
-    Bkj = zeros((N+1, m), dtype=an.dtype)
-    for row in range(1, m+1):
-        Bkj[row,:row] = -(an[:row])[::-1]
-    for row in range(m+1, N+1):
-        Bkj[row,:] = -(an[row-m:row])[::-1]
-    C = hstack((Akj, Bkj))
-    pq = linalg.solve(C, an)
-    p = pq[:n+1]
-    q = r_[1.0, pq[n+1:]]
-    return poly1d(p[::-1]), poly1d(q[::-1])
-
+    if m == 0:  # this is just the Taylor series
+        return _Polynomial(an), _Polynomial([1])
+    # first solve the Toeplitz system for q
+    # first row might contain tailing zeros
+    top = r_[an[n+1::-1][:m+1], [0]*(m-n-1)]
```

Comment:
Minor, trivial, optional: while we're at, how about spelling `np.r_` (and removing the old-fashioned `from numpy import r_`)